File: frontend/src/app/model/parameters/BookFilterParameters.ts
export class BookFilterParameters {
  name?: string;
  yearPublisher?: number;
  isbn?: BigInteger;
  publisher?: string;
  fromRating?: number;
  toRating?: number;
  authorNameAndSecondName?: string;


  constructor(name?: string, yearPublisher?: number, isbn?: BigInteger, publisher?: string, fromRating?: number, toRating?: number, authorNameAndSecondName?: string) {
    this.name = name;
    this.yearPublisher = yearPublisher;
    this.isbn = isbn;
    this.publisher = publisher;
    this.fromRating = fromRating;
    this.toRating = toRating;
    this.authorNameAndSecondName = authorNameAndSecondName;
  }
}

File: backend/src/main/java/com/softserve/utils/BookFilterParameters.java
package com.softserve.utils;

import lombok.Getter;
import lombok.Setter;

import java.math.BigInteger;

@Getter
@Setter
public class BookFilterParameters {

    private String name;
    private Integer yearPublisher;
    private BigInteger isbn;
    private String publisher;
    private Integer fromRating;
    private Integer toRating;
    private String searchingName;
    private BigInteger authorId;
}
